### DISCUSSION TITLE
Statically link antlr4-cpp-runtime to a console project

### DISCUSSION THREAD

**Discussion by uniquadev** [antlr/antlr4#3858] [Q&A] [answered]
Hi again ;,)

OS: Windows 11
IDE: Visual Studio Community 2022

I managed to compile the cpp-runtime in the dev branch, however I had problems linking the static library to my console project..

I tried making sure everything were fine:
* Include Path
* Linker Additional Input Path
* Project Reference
but without success :D

![devenv_gX2olvsIF4](https://user-images.githubusercontent.com/58218460/187945194-fc359e29-350c-4a8d-8207-12b599a7b1e7.png)

In order to keep this discussion simple I made a new solution with `antl4-runtime project` and a hello world `console project` with a simple `ANTLRInputStream` usage.

[Antlr4RuntimeTest.zip](https://github.com/antlr/antlr4/files/9471095/Antlr4RuntimeTest.zip)
[VirusTotal-Scan](https://www.virustotal.com/gui/file/0efcfe4d9747176db744e891cdf8a6982a9097624119e8a1b424bbda75f9edbd?nocache=1)

Expected problem:
![devenv_VmjoYnWaG6](https://user-images.githubusercontent.com/58218460/187948701-4f9aeac6-6d74-4289-883e-80513f024e96.png)

Similar discussion: https://github.com/antlr/antlr4/discussions/3356

**Comment by KvanTTT** [comment]
Could you please debug any test and see what files the test generates? Try debugging here: https://github.com/antlr/antlr4/blob/dev/runtime-testsuite/test/org/antlr/v4/test/runtime/cpp/CppRunner.java It supports both clang++ and msbuild.

You can run c++ tests using the following instruction: https://github.com/antlr/antlr4/blob/dev/doc/antlr-project-testing.md

**Comment by uniquadev** [comment, marked as answer]
I found the problem..

**Project Settings > Linker > Use Input Libraries** set to **Yes** (which by default is set to No)

> Also, I was trying to compile antlr runtime with another library so I ended up making them one single library avoiding errors
